scripts/generate_book_outline/generate_book_outline.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+const markdownExtension = ".md"
+
+func main() {
+	source := "site/book"
+	chapters := make([]chapter, 0)
+	chapterDirs, err := ioutil.ReadDir(source)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+
+	for _, dir := range chapterDirs {
```

Comment:
it's not clear what this block does and it probably could make it easier on the reader to split it up and have smaller ones that also have comments about what they do.

Reply:
Separated and commented.